### PR TITLE
implement lwm2m deviceshifu and gateway

### PR DESCRIPTION
**Pull request by rhoninl**
**What this PR does / why we need it**:
implement deviceshifu lwm2m and gateway lwm2m
**Will this PR make the community happier**? 
yes
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [ ] unit test
- [ ] e2e test
- [ ] other (please specify)

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2m.go [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+							logger.Infof("Observe data: %v", data)
+							// TODO need to push data to telemetry service
+						})
+					})
+				}
+			}
+		default:
+			logger.Errorf("EdgeDevice protocol %v not supported in deviceShifu_http_http", protocol)
+			return nil, errors.New("wrong protocol not supported in deviceShifu_http_http")
+		}
```
Comment (rhoninl) [inline pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2m.go:95]: need to update log

File: pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2m.go [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+)
+
+const (
+	DeviceNameHeaderField = "Device-Name"
+)
+
+// New This function creates a new Device Shifu based on the configuration
+func New(deviceShifuMetadata *deviceshifubase.DeviceShifuMetaData) (*DeviceShifuLwM2M, error) {
+	if deviceShifuMetadata.Namespace == "" {
+		return nil, fmt.Errorf("DeviceShifuHTTP's namespace can't be empty")
```
Comment (rhoninl) [inline pkg/deviceshifu/deviceshifulwm2m/deviceshifulwm2m.go:42]: need to update

**Comment by tomqin93** [comment]
@rhoninl you can close this PR now

**Comment by rhoninl** [comment]
finished in #1001 and #1002

**Pull request closed without merging**